---
author: https://github.com/umccr/linxreport
date: "`r format(Sys.time())`"
output:
  html_document:
    theme: cosmo
    css: style.css
    code_download: true
  rmdformats::material:
    highlight: kate
params:
  title: "LINX Reporter"
  sample: "COLO829v003T"
  plot_dir: !expr system.file("extdata/plots", package = "linxreport")
  table_dir: !expr system.file("extdata/tables", package = "linxreport")
description: "LINX Reporter"
title: "`r paste(params$sample, params$title)`"
---

```{r knitr_opts, include=F}
knitr::opts_chunk$set(
  collapse = TRUE, echo = FALSE,
  warning = FALSE, message = FALSE,
  error = FALSE
)
```

```{r load_pkgs, message=FALSE, warning=FALSE}
library(details, include.only = "details")
library(dplyr, include.only = c("select", "mutate", "filter", "arrange"))
library(DT, include.only = "datatable")
library(glue, include.only = "glue")
library(linxreport)
library(stringr, include.only = "str_match")
```

```{r funcs}
blank_lines <- function(n = 10) {
  cat(rep("&nbsp;  ", n), sep = "\n")
}

# path to table
ptab <- function(type) {
  x <- file.path(params$table_dir, glue("{params$sample}.linx.{type}.tsv"))
  x
}
```

```{r read_tables}
tabs <- list(
  main_svs = ptab("svs") |> linxreport::linx_svs_process(),
  main_breakend = ptab("breakend") |> linxreport::linx_breakend_process(),
  main_clusters = ptab("clusters") |> linxreport::linx_clusters_process(),
  main_links = ptab("links") |> linxreport::linx_links_process(),
  main_fusion = ptab("fusion") |> linxreport::linx_fusion_process(),
  main_drivercatalog = ptab("driver.catalog") |> linxreport::linx_drivercatalog_process(),
  main_drivers = ptab("drivers") |> linxreport::linx_drivers_process(),
  vis_copy_number = ptab("vis_copy_number") |> linxreport::linx_viscopynumber_read(),
  vis_fusion = ptab("vis_fusion") |> linxreport::linx_visfusion_read(),
  vis_gene_exon = ptab("vis_gene_exon") |> linxreport::linx_visgeneexon_read(),
  vis_protein_domain = ptab("vis_protein_domain") |> linxreport::linx_visproteindomain_read(),
  vis_segments = ptab("vis_segments") |> linxreport::linx_vissegments_read(),
  vis_sv_data = ptab("vis_sv_data") |> linxreport::linx_vissvdata_read()
)
```

## Tables {.tabset #tables}

### Main {.tabset}

```{r svs0}
l <- tabs$main_svs
```

#### Annotations (`r nrow(l$tab)`)

- additional annotations of each non PON filtered break junction
- `svs.tsv`

```{r svs}
details::details(knitr::kable(l$descr), lang = "none", summary = "Column Description")
dt_table(l$tab)
```

```{r blank2, results='asis'}
blank_lines(2)
```

```{r breakend0}
l <- tabs$main_breakend
```

#### Breakends (`r nrow(l$tab)`)

- impact of each non PON filtered break junction on each overlapping gene
- `breakend.tsv`

```{r breakend}
details::details(knitr::kable(l$descr), lang = "none", summary = "Column Description")
dt_table(l$tab)
```

```{r clusters0}
l <- tabs$main_clusters
```

#### Clusters (`r nrow(l$tab)`)

- clustering of all non PON filtered SV events and their resolved
  classification.
- `clusters.tsv`

```{r clusters}
details::details(knitr::kable(l$descr), lang = "none", summary = "Column Description")
dt_table(l$tab)
```

```{r links0}
l <- tabs$main_links
```

#### Links (`r nrow(l$tab)`)

- segments joining break junction pairs predicted to be linked and phased in
  _cis_ on the derivative chromosome.
- `links.tsv`

```{r links}
details::details(knitr::kable(l$descr), lang = "none", summary = "Column Description")
dt_table(l$tab)
```

```{r fusion0}
l <- tabs$main_fusion
```

#### Fusions (`r nrow(l$tab)`)

- All inframe and outframe fusions predicted in the sample including HMF
  fusion knowledgebase annotations.
- `fusion.tsv`

```{r fusion}
details::details(knitr::kable(l$descr), lang = "none", summary = "Column Description")
dt_table(l$tab)
```

```{r drivercatalog0}
l <- tabs$main_drivercatalog
```

#### Driver Catalog (`r nrow(l$tab)`)

- Reproduction of the driver catalog produced by PURPLE with homozygous
  disruption events appended.
- `driver.catalog.tsv`

```{r drivercatalog}
details::details(knitr::kable(l$descr), lang = "none", summary = "Column Description")
dt_table(l$tab)
```

```{r drivers0}
l <- tabs$main_drivers
```

#### Drivers (`r nrow(l$tab)`)

- Linkage of drivers from driver catalog to SV cluster which contributed to
  those drivers including LOH, deletion, disruption and amplification events.
- `drivers.tsv`

```{r drivers}
details::details(knitr::kable(l$descr), lang = "none", summary = "Column Description")
dt_table(l$tab)
```

### Vis {.tabset}

```{r vis_copy_number0}
l <- tabs$vis_copy_number
```

#### Copy Number (`r nrow(l)`)

- `vis_copy_number.tsv`

```{r vis_copy_number}
dt_table(l) |>
  DT::formatCurrency(~ Start + End, currency = "", interval = 3, mark = ",", digits = 0)
```

```{r blank1, results='asis'}
blank_lines(2)
```

```{r vis_fusion0}
l <- tabs$vis_fusion
```

#### Fusion (`r nrow(l)`)

- `vis_fusion.tsv`

```{r vis_fusion}
dt_table(l)
```

```{r vis_gene_exon0}
l <- tabs$vis_gene_exon
```

#### Gene Exon (`r nrow(l)`)

- `vis_gene_exon.tsv`

```{r vis_gene_exon}
dt_table(l) |>
  DT::formatCurrency(~ ExonStart + ExonEnd, currency = "", interval = 3, mark = ",", digits = 0)
```

```{r vis_protein_domain0}
l <- tabs$vis_protein_domain
```

#### Protein Domain (`r nrow(l)`)

- `vis_protein_domain.tsv`

```{r vis_protein_domain}
dt_table(l) |>
  DT::formatCurrency(~ Start + End, currency = "", interval = 3, mark = ",", digits = 0)
```

```{r vis_segments0}
l <- tabs$vis_segments
```

#### Segments (`r nrow(l)`)

- `vis_segments.tsv`

```{r vis_segments}
dt_table(l)
```

```{r vis_sv_data0}
l <- tabs$vis_sv_data
```

#### SV Data (`r nrow(l)`)

- `vis_sv_data.tsv`

```{r vis_sv_data}
dt_table(l) |>
  DT::formatCurrency(~ PosStart + PosEnd, currency = "", interval = 3, mark = ",", digits = 0)
```

## Plots (Chromosome) {.tabset #plots-chrom}

```{r plot_chrom, results='asis'}
chrom_plots <- linxreport::linx_path_plot_chr(params)
if (!is.null(chrom_plots)) {
  cat(glue("\n**Total:** {nrow(chrom_plots)} chromosome plots"), "\n\n\n")
  for (i in seq_len(nrow(chrom_plots))) {
    r <- chrom_plots[i, ]
    cat(glue("\n### chr{r$chrom}"), "\n")
    cat(glue("\nnsegs: {as.integer(r$nsegs)}"), "\n\n\n")
    cat(glue("<img src=\"{r$path}\" height=\"1000\"/>"), "\n\n\n")
    blank_lines(1)
    cat("\n\n\n")
  }
} else {
  cat("No LINX chromosome plots were generated")
}
```

## Plots (Cluster) {.tabset #plots-cluster}

```{r plot_cluster, results='asis'}
cluster_plots <- linxreport::linx_path_plot_cluster(params)
if (!is.null(cluster_plots)) {
  cat(glue("\n**Total:** {nrow(cluster_plots)} cluster plots"), "\n\n\n")
  for (i in seq_len(nrow(cluster_plots))) {
    r <- cluster_plots[i, ]
    cat(glue("\n### {r$clusterId}"), "\n\n\n")
    cat(glue(
      "clusterId: {r$clusterId} | ",
      "svId: {r$svId} | nsegs: {r$nsegs}"
    ), "\n\n\n")
    cat(glue("<img src=\"{r$path}\" height=\"1000\"/>"), "\n\n\n")
    blank_lines(2)
    cat("\n\n\n")
  }
} else {
  cat("No LINX cluster plots were generated")
}
```

## Addendum {.tabset #addendum}

- `{linxreport}` documentation: https://umccr.github.io/linxreport/
- `{linxreport}` issues: https://github.com/umccr/linxreport/issues
- LINX documentation: https://github.com/hartwigmedical/hmftools/tree/master/linx
- LINX issues: https://github.com/hartwigmedical/hmftools/issues

For R session information and input parameters for this report, click below.

<details>
<summary>Click</summary>


### Session Info - Packages

```{r session_info1}
si <- linxreport::session_info_tbls()
si$si_pkg |> knitr::kable()
```

### Session Info - Platform

```{r session_info2}
si$si_pl |> knitr::kable()
```

### Report Inputs

```{r report_inputs}
report_inputs <- dplyr::tibble(key = names(params), value = unlist(params))
report_inputs |> knitr::kable()
```

</details>
